detectron2/utils/events.py

```diff
@@ -484,3 +492,41 @@ def clear_histograms(self):
         This should be called after histograms are written to tensorboard.
         """
         self._histograms = []
+
+
+class WandbWriter(EventWriter):
+    """
+    Write all scalars to a wandb tool.
+    """
+
+    def __init__(self, project: str='detectron2', config: Union[Dict, CfgNode] = {}, window_size: int = 20, **kwargs):
+        """
+        Args:
+            project (str): W&B Project name
+            config Union[Dict, CfgNode]: the project level configuration object
+            window_size (int): the scalars will be median-smoothed by this window size
+
+            kwargs: other arguments passed to `wandb.init(...)`
+        """
+        if wandb is None:
+            raise Exception("WandbWriter requires wandb. please install using `pip install wandb`")
+
+        self._window_size = window_size
+        self._run = wandb.init(
+            project=project,
+            config=config,
+            **kwargs
+        )
```

Comment:
Like evaluator I think it also needs to check if `wandb.run` is already set? Maybe user already call `wandb.init` before training - that seems to be a common pattern if I'm not mistaken

Reply:
Oh yep. Good catch. Thanks!

Reply:
done!